webui/react/src/utils/data.test.ts

```diff
@@ -136,6 +141,58 @@ describe('data utility', () => {
     });
   });
 
+  describe('chained object manipulators', () => {
+    let config = clone(testConfig);
+
+    beforeAll(() => {
+      config = clone(testConfig);
+    });
+
+    describe('getSubObject', () => {
+      it('should return undefined for bad paths', () => {
+        const actual = getSubObject(config, [ 'x', 'y', 'z' ]);
+        expect(actual).toBeUndefined();
+      });
+
```

Comment:
**suggestion:** add a test where it returns undefined even when it matches part of the path. See the comment below about `getSubObject` it may return up to what it finds rather than a definitive `undefined` if there's partial matching but not a complete match.